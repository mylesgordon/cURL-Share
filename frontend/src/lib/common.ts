export function splitAndTrim(input: string): Array<string> {
		return input.split(',').map((item) => item.trim());
	}