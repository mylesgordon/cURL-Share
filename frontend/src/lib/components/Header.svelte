<script lang="ts">
	import { Header, HeaderNav, HeaderNavItem } from 'agnostic-svelte';
</script>

<Header>
	<a href="/">cURL Share</a>
	<HeaderNav>
		<HeaderNavItem>
			<a href="/log-in">Log In</a>
		</HeaderNavItem>
	</HeaderNav>
</Header>
