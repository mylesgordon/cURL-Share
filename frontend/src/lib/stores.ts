import { readable, writable } from 'svelte/store';

export const backendUrl = readable(import.meta.env.VITE_BACKEND_BASE_URL);
export const isLoggedIn = writable(false);
