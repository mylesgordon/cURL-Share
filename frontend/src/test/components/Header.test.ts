import { isLoggedIn } from '$lib/stores';
import { render } from '@testing-library/svelte';
import { testMatchingSnapshot } from '../common';
import Header from '$lib/components/Header.svelte';

describe('Header component', () => {
	let isLoggedInValue: boolean;
	isLoggedIn.subscribe((value) => {
		isLoggedInValue = value;
	});

	testMatchingSnapshot(Header, { isLoggedInResponse: false });
	testMatchingSnapshot(Header, { isLoggedInResponse: true });

	it('should update isLoggedIn store with correct value', () => {
		const testValues = [false, true];
		testValues.forEach((testValue) => {
			render(Header, { isLoggedInResponse: testValue });
			expect(isLoggedInValue).toBe(testValue);
		});
	});
});
