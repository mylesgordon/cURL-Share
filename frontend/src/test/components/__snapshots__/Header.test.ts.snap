// Vitest Snapshot v1

exports[`Header component > should match snapshot 1`] = `
<body>
  <div>
    <nav
      class="header s-0ZVLjeDTECqc"
    >
      <div
        class="header-content s-0ZVLjeDTECqc"
      >
         
        <a
          href="/"
        >
          cURL Share
        </a>
         
        <nav
          class=" s-IJVakqdrdFix"
        >
          <ul
            class="header-nav s-IJVakqdrdFix"
          >
            <li
              class="header-nav-item s-eWnZjIejw3S-"
            >
              <a
                href="/log-in"
              >
                Log In
              </a>
              
            </li>
            <!--&lt;HeaderNavItem&gt;-->
          </ul>
        </nav>
        <!--&lt;HeaderNav&gt;-->
         
      </div>
    </nav>
    <!--&lt;Header&gt;-->
    <!--&lt;Header&gt;-->
  </div>
</body>
`;

exports[`Header component > should match snapshot 2`] = `
<body>
  <div>
    <nav
      class="header s-0ZVLjeDTECqc"
    >
      <div
        class="header-content s-0ZVLjeDTECqc"
      >
         
        <a
          href="/"
        >
          cURL Share
        </a>
         
        <nav
          class=" s-IJVakqdrdFix"
        >
          <ul
            class="header-nav s-IJVakqdrdFix"
          >
            <li
              class="header-nav-item s-eWnZjIejw3S-"
            >
              <a
                href="/log-out"
              >
                Log Out
              </a>
              
            </li>
            <!--&lt;HeaderNavItem&gt;-->
          </ul>
        </nav>
        <!--&lt;HeaderNav&gt;-->
         
      </div>
    </nav>
    <!--&lt;Header&gt;-->
    <!--&lt;Header&gt;-->
  </div>
</body>
`;
